app/src/main/java/io/halkyon/resource/page/ApplicationResource.java

```diff
@@ -108,6 +108,9 @@ public Response doClaimApplication(@PathParam("id") long applicationId, @FormPar
             throw new NotAcceptableException(String.format("Claim %s has no services available", claimId));
         }
         if (claim.service.installable) {
+            if (claim.application == null) {
+                throw new NotFoundException("Claim with name " + claim.name + " is not associated to an application.");
```

Comment:
Will the HTTP request receive in this case a HTTP 500 error ?

Reply:
Yes, I think so